cabal-install/src/Distribution/Client/CmdBuild.hs

```diff
@@ -95,19 +97,16 @@ defaultBuildFlags = BuildFlags
 -- "Distribution.Client.ProjectOrchestration"
 --
 buildAction :: NixStyleFlags BuildFlags -> [String] -> GlobalFlags -> IO ()
-buildAction flags@NixStyleFlags { extraFlags = buildFlags, ..} targetStrings globalFlags = do
+buildAction flags@NixStyleFlags { extraFlags = buildFlags, ..} targetStrings globalFlags
+  = withContextAndSelectors RejectNoTargets "" Nothing flags targetStrings globalFlags $ \targetCtx ctx targetSelectors -> do
     -- TODO: This flags defaults business is ugly
     let onlyConfigure = fromFlag (buildOnlyConfigure defaultBuildFlags
                                  <> buildOnlyConfigure buildFlags)
         targetAction
             | onlyConfigure = TargetActionConfigure
             | otherwise = TargetActionBuild
 
-    baseCtx <- establishProjectBaseContext verbosity cliConfig OtherCommand
-
-    targetSelectors <-
-      either (reportTargetSelectorProblems verbosity) return
-      =<< readTargetSelectors (localPackages baseCtx) Nothing targetStrings
+    baseCtx <- updateContextAndWriteScriptProjectFiles ctx targetCtx
```

Comment:
Maybe i am missing something, this seems to be executed for all targets but the code is in ScriptUtils. If the code can handle other targets than script files maybe should be more generic in the name and place (or make it more concrete)

Reply:
updateContextAndWriteScriptProjectFiles uses the targetCtx passed to it to only write in the script case. The name is a little misleading, but I thought I mentioned it in the doc string.